### PR TITLE
Issue12 product page listing part2

### PR DESCRIPTION
**Pull request by ninetteadhikari**
- Adds material design styling to card
- Adds filter by price range, brand name, availability date
- Changes Product model "name" property to "title"

### CODE REVIEW

File: client/src/components/ProductsList.js [hunk 1]
```diff
@@ -65,24 +80,45 @@ export default class ProductsList extends Component {
...
-      return searchMatched && categoryMatched;
+      let priceMatched =
+        product.price >= this.state.priceValue[0] &&
+        product.price <= this.state.priceValue[1];
+
+      // Date.parse converts the date string into milliseconds
+      let dateMatched =
+        Date.parse(product.availability) <= Date.parse(this.state.selectedDate);
+
+      return searchMatched && categoryMatched && priceMatched && dateMatched;
```
Comment (1000miles) [inline client/src/components/ProductsList.js:118]: @ninetteadhikari As discussed yesterday if there is time could you also please add filtering by `brand` and `location`?

File: client/src/components/ProductsList.js [hunk 2]
```diff
@@ -2,23 +2,29 @@ import React, { Component } from "react";
 import axios from "axios";
 import { Link } from "react-router-dom";
 import SearchFilter from "./SearchFilter";
+import useStyles from "./ProductListStyles";
+import {
+  Typography,
+  TextField,
+  Grid,
+  Box,
```
Comment (1000miles) [inline client/src/components/ProductsList.js]: Are we going to use this on thee ProductListing page? If not, let's remove it later. :)

```
  Line 8:   'TextField' is defined but never used  no-unused-vars
  Line 9:   'Grid' is defined but never used       no-unused-vars
  Line 10:  'Box' is defined but never used        no-unused-vars
```

File: client/src/components/ProductsList.js [hunk 3]
```diff
@@ -46,16 +52,25 @@ export default class ProductsList extends Component {
...
+  };
 
-    const classes = useStyles;
+  handleDateChange = date => {
+    this.setState({
+      selectedDate: date
+    });
+  };
+
+  render() {
```
Comment (1000miles) [inline client/src/components/ProductsList.js:67]: @ninetteadhikari I am getting the fake data in the backend via postman for `http://localhost:3000/api/products`. I can see the page is rendered with console.log(`RENDER [ProductListing.js]...`) but I can't get the props rendered. Am I missing something?

```json
 {
        "location": {
            "latitude": -18.0912,
            "longitude": 70.6121
        },
        "imageUrl": [
            "http://lorempixel.com/640/480/business",
            "http://lorempixel.com/640/480/business",
            "http://lorempixel.com/640/480/business",
            "http://lorempixel.com/640/480/business",
            "http://lorempixel.com/640/480/business"
        ],
        "tags": [
            "tablets",
            "chair",
            "kitchen",
            "cabinet",
            "reception"
        ],
        "wishlist": [
            "5d97c695eabd3708476fc8f7",
            "5d97c695eabd3708476fc8f2",
            "5d97c695eabd3708476fc8f0"
        ],
        "requested": [
            "5d97c695eabd3708476fc8f2",
            "5d97c695eabd3708476fc8f2"
        ],
        "_id": "5d97c8c21abc6708eb9eca53",
        "title": "Aut fuga sint nemo.",
        "description": "Ex amet nesciunt dolore eum sint. Sequi et non unde explicabo provident veniam voluptas unde beatae. Molestiae nulla est eos nobis.",
        "brand": "Madame Stoltz",
        "category": "Carpentry",
        "quantity": 8,
        "price": 131.22,
        "currency": "JMD",
        "company": "Thompson - Feeney",
        "availability": "2020-02-14T07:25:00.168Z",
        "warrantyUntil": "2020-08-04T06:02:53.707Z",
        "condition": "used",
        "isSold": false,
        "seller": "5d97c695eabd3708476fc8f3",
        "buyer": "5d97c695eabd3708476fc8f7",
        "__v": 0
    }
```

**Review by 1000miles — COMMENTED** [review]
Great work, @ninetteadhikari ! 🍬 I left some comments.

File: client/src/components/ProductsList.js (hunk 1 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js:118]: cool! will add it to 'part3' branch

File: client/src/components/ProductsList.js (hunk 2 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js]: i'll remove it to keep it clean:)

File: client/src/components/ProductsList.js (hunk 1 above)
Comment (1000miles) [reply client/src/components/ProductsList.js:118]: Great! :)

File: client/src/components/ProductsList.js (hunk 2 above)
Comment (1000miles) [reply client/src/components/ProductsList.js]: 👍

File: client/src/components/ProductsList.js (hunk 3 above)
Comment (1000miles) [reply client/src/components/ProductsList.js:67]: Fixed. :D

**Review by 1000miles — APPROVED** [review]
Thanks a lot! 🎉

**Pull request merged**